### PR TITLE
Seperate e2e test for addons

### PR DESCRIPTION
**Pull request by Revolyssup**
Signed-off-by: ashish <ashishjaitiwari15112000@gmail.com>

**Description**

This PR fixes #

**Notes for Reviewers**

**[Signed commits](../CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

File: .github/workflows/e2etest-servicemeshandaddon.yaml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+          path: ./.github/install/deploywithaddons.yaml 
+
+  TestIstio:
+    needs: SetPatterfile
+    uses: meshery/meshery/.github/workflows/testadapters.yaml@master
+    with:
+      expected_pods: istiod,istio-egressgateway,istio-ingressgateway,grafana,prometheus
+      expected_pods_namespaces: istio-system,istio-system,istio-system,istio-system,istio-system
+      deployment_url: https://raw.githubusercontent.com/meshery/meshery/master/install/deployment_yamls/k8s/meshery-istio-deployment.yaml
+      service_url: https://raw.githubusercontent.com/meshery/meshery/master/install/deployment_yamls/k8s/meshery-istio-service.yaml
```
Comment (leecalcote) [inline .github/workflows/e2etest-servicemeshandaddon.yaml:42]: We're trying to get away from using these and moving to helm. // @DelusionalOptimist

File: .github/install/deploywithaddons.yaml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+        namespaces:
+          - default
+  grafana:
+    type: GrafanaIstioAddon
+    namespace: istio-system
+    dependsOn:
+      - istio
+      - prometheus
+
+  prometheus:
```
Comment (leecalcote) [inline .github/install/deploywithaddons.yaml:20]:
```suggestion
  metrics:
```

File: .github/install/deploywithaddons.yaml [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+  istio:
+    type: IstioMesh
+    namespace: istio-system
+    settings:
+      version: #will be filled dynamically in workflow
+    traits:
+      automaticSidecarInjection:
+        namespaces:
+          - default
+  grafana:
```
[the remark this answers is not in the document]
Comment (leecalcote) [reply .github/install/deploywithaddons.yaml:13]: Is it possible to make the names of these generic?

File: .github/workflows/e2etest-servicemeshinstall.yaml [hunk 4]
```diff
@@ -1,4 +1,4 @@
-name: Test Istio Adapter
+name: Test Istio Adapter - control plane
```
Comment (leecalcote) [inline .github/workflows/e2etest-servicemeshinstall.yaml:1]:
```suggestion
name: Test Meshery Adapter for Istio - control plane 
```

File: .github/install/deploywithaddons.yaml (hunk 3 above)
[the remark this answers is not in the document]
Comment (Revolyssup) [reply .github/install/deploywithaddons.yaml:13]: Both the service name as well as type?

**Review by leecalcote — APPROVED** [review]

**Pull request merged**